### PR TITLE
langchain[minor]: Couchbase document loader

### PR DESCRIPTION
**Pull request by lokesh-couchbase**
Description:
Adds the document loader for [Couchbase](http://couchbase.com/), a distributed NoSQL database.
Dependencies:
Added the Couchbase SDK as an optional dependency.

### CODE REVIEW

File: docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        SELECT h.* FROM `travel-sample`.inventory.hotel h 
+        WHERE h.country = 'United States'
+        LIMIT 1
+    `;
+```
+
+### Connect to Couchbase Cluster
+
+```typescript
+  const couchbaseClient = await Cluster.connect(connectionString, {
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx:45]: Formatting is a bit off here - you could also write this as a Deno notebook if you'd like?

https://github.com/langchain-ai/langchainjs/blob/main/docs/core_docs/docs/expression_language/streaming.ipynb

File: docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import { CouchbaseDocumentLoader } from "langchain/document_loaders/web/couchbase";
+import { Cluster } from "couchbase"
+
+const connectionString = "couchbase://localhost";  // valid couchbase connection string
+const dbUsername = "Administrator"  // valid database user with read access to the bucket being queried
+const dbPassword = "Password"  // password for the database user
+
+// query is a valid SQL++ query
+const query = `
+        SELECT h.* FROM `travel-sample`.inventory.hotel h
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx:36]: Fix formatting

**Comment by jacoblee93** [comment]
LGTM, just fix formatting in docs please!

File: langchain/src/document_loaders/tests/couchbase.int.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    configProfile: "wanDevelopment"
+  });
+  const loader = new CouchbaseDocumentLoader(
+    couchbaseClient,
+    query,
+    validPageContentFields,
+    validMetadataFields
+  );
+  const docs = await loader.load();
+  console.log(docs);
```
Comment (JakubKontra) [inline langchain/src/document_loaders/tests/couchbase.int.test.ts:29]:
```suggestion
  console.log(docs);
```
I'd remove console.log

File: langchain/src/document_loaders/tests/couchbase.int.test.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  expect(docs.length).toBeGreaterThan(0);
+
+  for (const doc of docs) {
+    console.log(doc);
```
Comment (JakubKontra) [inline langchain/src/document_loaders/tests/couchbase.int.test.ts:33]: Same here - remove console.log

File: docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx (hunk 1 above)
Comment (lokesh-couchbase) [reply docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx]: I have fixed the formatting

File: docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx (hunk 2 above)
Comment (lokesh-couchbase) [reply docs/core_docs/docs/integrations/document_loaders/web_loaders/couchbase.mdx]: fixed formatting

File: langchain/src/document_loaders/tests/couchbase.int.test.ts (hunk 3 above)
Comment (lokesh-couchbase) [reply langchain/src/document_loaders/tests/couchbase.int.test.ts]: Removed logging

File: langchain/src/document_loaders/tests/couchbase.int.test.ts (hunk 4 above)
Comment (lokesh-couchbase) [reply langchain/src/document_loaders/tests/couchbase.int.test.ts]: Removed logging

**Comment by lokesh-couchbase** [comment]
Thanks for the comments and I have updated the PR accordingly.

@jacoblee93 one thing I was facing issue with building docs from about 3 days. It appears to be issue from .ipynb type docs. I have tried to detail it on discord [Javascript-Contributing](https://discord.com/channels/1038097195422978059/1076183245479215104) channel. A quick look from your end may help it out

**Comment by jacoblee93** [comment]
Thank you! Still looks to be failing format but I'll fix

**Pull request merged**